### PR TITLE
Deprecate 1l and lint 0755

### PR DESCRIPTION
**Pull request by som-snytt**
Advise to use 1L.

Octal syntax error is demoted to lint warning
which can be disabled by power number users,
who can freely write decimals literals with
leading zero.

Squash/rebase of https://github.com/scala/scala/pull/7388

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/parser/Scanners.scala [hunk 1]
```diff
@@ -1096,12 +1094,17 @@ trait Scanners extends ScannersCommon {
         nextChar()
         getFraction()
       }
+      // 1l is an acknowledged bad practice
```
Comment (He-Pin) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:1097]: Great!

File: test/files/neg/literals.scala [hunk 2]
```diff
@@ -45,3 +44,10 @@ trait MoreSadness {
 
   def twoHuge: Double    = { 1.7976931348623159e308 } // double precision floating point number too large
 }
+
+trait Lengthy {
+
+  def bad = 1l
+
+  def worse = 123l
```
Comment (adriaanm) [inline test/files/neg/literals.scala:52]: I would think `32l` is worstest.

**Review by adriaanm — APPROVED** [review]

File: test/files/neg/literals.scala (hunk 2 above)
Comment (adriaanm) [reply test/files/neg/literals.scala:52]: or is it worcestershire?

**Comment by adriaanm** [comment]
I guess you could say something about 1igatures and lDEs and 1inters, but l agree we don't kneed to confuse. (Ok, I've been staring down the 🐰 🕳 too long today.)

File: test/files/neg/literals.scala (hunk 2 above)
Comment (som-snytt) [reply test/files/neg/literals.scala:52]: L01, you're right!

**Comment by SethTisue** [comment]
but what about Polish `ł`?

(which in the monospace font in my browser, anyway, kind of looks like a meerkat scanning for predators)

**Comment by som-snytt** [comment]
For Polish notation, we recommend uppercase in honor of Łukasiewicz.

**Comment by SethTisue** [comment]
@som-snytt is something wrong here that I don't get a "warning: there was one deprecation warning" meta-warning in the REPL? it doesn't match the behavior of ordinary method deprecations, as shown in this transcript:

```
Welcome to Scala 2.13.0-pre-f1dec97 [...]

scala> (new Thread).stop()
warning: there was one deprecation warning; for details, enable `:setting -deprecation' or `:replay -deprecation'

scala> 1l
res1: Long = 1

scala> :reset -deprecation
Resetting interpreter state. [...]

scala> 1l
        ^
       warning: Lowercase el for long is not recommended because it is easy to confuse with numeral 1; use uppercase L instead
res0: Long = 1
```

this came up over at https://github.com/scala/scala/pull/7458, where @lrytz's PR had `1l` in it and @Ichoran noticed

in scala/scala we don't have `-deprecation` enabled because it would produce an enormous amount of noise, in which other warnings would drown. (and our build actually doesn't have many other warnings, these days.)

**Comment by som-snytt** [comment]
@SethTisue the refactored REPL doesn't show parsing deprecations; there is a ticket, but I haven't addressed it.

**Comment by SethTisue** [comment]
oh ok, it's https://github.com/scala/bug/issues/11402, thanks

**Pull request merged**